src/components/profile/CircleButtonWrapper.js

```diff
@@ -18,7 +18,8 @@ const CircleButtonWrapper = ({
   iconColor,
   iconSize,
 }) => {
-  const _onPress = useOnPress(onPress)
+  const onIconPress = useOnPress(onPress || noop, [onPress])
+  const _onPress = onPress ? onIconPress : noop
```

Comment:
firstly the conditions || and ? seem redudant.
secondly can pass _onPress undefined is no onPress was passed? because a default noop onPress will result in changing the pointer of the mouse over on the icons, making the user think it is clickable even though its just "noop"

Reply:
I think the better way will be to adjust useOnPress. The hook shouldn't return a function if the callback is empty:

```javascript
import { useCallback, useRef } from 'react'
import { isFunction, isNil } from 'lodash'

const useOnPress = (callback, deps = []) => {
  const wrappedCallback = useCallback(
    event => {
      if (event && isFunction(event.preventDefault)) {
        event.preventDefault()
      }

      return callback(event)
    },
    [callback, ...deps],
  )

  if (callback) {
    return wrappedCallback
  }
}
  
export const useDebouncedOnPress = (callback, deps = []) => {
  const nextInvokationRef = useRef(null)

  const debouncedCallback = useMemo(() => {
    if (!callback) {
      return
    }

    return event => {
      const nextInvokation = nextInvokationRef.current
      const currentTs = Date.now()

      if (isNil(nextInvokation) || currentTs >= nextInvokation) {
        callback(event)
      }

      nextInvokationRef.current = currentTs + 500
    }
  }, [callback])
  
  return useOnPress(debouncedCallback, deps)
}

export default useOnPress
```

then you could just use:

```javascript
const onIconPress = useOnPress(onPress)
```

This also simplifies the Icon component (you could try & check this code):

```javascript
export default withTheme(({ theme, color, size = 16, onPress, ...iconProps }: IconProps) => {
  const onIconPress = useOnPress(onPress)

  const iconColor = useMemo(() => {
    const { colors } = theme

    return colors[color] || color || colors.primary
  }, [theme, color])

  return <Icon onPress={onIconPress} size={size} color={iconColor} {...iconProps} />
})
```